tests/phpunit/tests/user/wp-user/getUserBy.php

```diff
@@ -0,0 +1,194 @@
+		self::$user_id = $factory->user->create( self::$user_data );
+	}
+
+	/**
+	 * @dataProvider data_id_field_for_valid_userid
+	 */
+	public function test_id_field_for_valid_userid( $cast_func ) {
+		$user_id = $cast_func( self::$user_id );
+
+		$user = WP_User::get_data_by( 'id', $user_id );
+		$this->assertInstanceOf( 'stdClass', $user, 'User is not an instance of stdClass' );
+		$this->assertSame( self::$user_id, (int) $user->ID, 'User ID does not match the user generated by the factory' );
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array
+	 */
+	public function data_id_field_for_valid_userid() {
+		return array(
+			'integer user ID' => array( 'intval' ),
+			'string user ID'  => array( 'strval' ),
+			'float user ID'   => array( 'floatval' ),
+		);
+	}
+
+	/**
+	 * @dataProvider data_id_field_for_invalid_userid
+	 * @group        me
+	 */
+	public function test_id_field_for_invalid_userid( $user_id ) {
+		$this->assertFalse( WP_User::get_data_by( 'id', $user_id ) );
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array
+	 */
+	public function data_id_field_for_invalid_userid() {
+		return array(
+			'integer'             => array( 99999 ),
+			'integer less than 0' => array( -1 ),
+			'integer 0'           => array( 0 ),
+			'non-numeric string'  => array( 'non-numeric' ),
+			'string integer'      => array( '99999' ),
+			'string less than 0'  => array( '-1' ),
+			'string 0'            => array( '0' ),
+		);
+	}
+
+	public function test_alias_of_id() {
+		$user = WP_User::get_data_by( 'ID', self::$user_id );
+		$this->assertInstanceOf( 'stdClass', $user, 'User is not an instance of stdClass' );
+		$this->assertSame( self::$user_id, (int) $user->ID, 'User ID does not match the user generated by the factory' );
+	}
+
+	/**
+	 * @dataProvider data_scalar_value
+	 */
+	public function test_scalar_value( $field, $value ) {
+		if ( 'id' === $field || 'ID' === $field ) {
+			$value = self::$user_id;
+		}
+
+		$user = WP_User::get_data_by( $field, $value );
+		$this->assertInstanceOf( 'stdClass', $user, 'User is not an instance of stdClass' );
+		$this->assertSame( self::$user_id, (int) $user->ID, 'User ID does not match the user generated by the factory' );
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array
+	 */
+	public function data_scalar_value() {
+		return array(
+			'id'          => array(
+				'field' => 'id',
+				'value' => null,
+			),
+			'ID'          => array(
+				'field' => 'id',
+				'value' => null,
+			),
+			'slug'        => array(
+				'field' => 'slug',
+				'value' => self::$user_data['user_nicename'],
+			),
+			'email'       => array(
+				'field' => 'email',
+				'value' => self::$user_data['user_email'],
+			),
+			'login: null' => array(
+				'field' => 'login',
+				'value' => self::$user_data['user_login'],
+			),
+		);
+	}
+
+	/**
+	 * @dataProvider data_nonscalar_value
+	 */
+	public function test_nonscalar_value( $field, $value ) {
```

Comment:
```suggestion
	 * @dataProvider data_nonscalar_value
	 *
	 * @param string $field The field to query against.
	 * @param mixed  $value A non-scalar field value.
	 */
	public function test_nonscalar_value( $field, $value ) {
```
Parameters from data providers should be documented.